subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/ExtensionContainerExtensions.kt

```diff
@@ -42,18 +43,40 @@ operator fun ExtensionContainer.get(name: String): Any =
  * If the extension is found but cannot be cast to the expected type it will throw an [IllegalStateException].
  *
  * @param name extension name
- * @return extension, never null
+ * @return extension, never `null`
  * @throws [UnknownDomainObjectException] When the given extension is not found.
  * @throws [IllegalStateException] When the given extension cannot be cast to the expected type.
  */
 @Suppress("extension_shadowed_by_member")
-inline fun <reified T : Any> ExtensionContainer.getByName(name: String) =
-    getByName(name).let {
-        it as? T
-            ?: throw IllegalStateException(
-                "Element '$name' of type '${it::class.java.name}' from container '$this' cannot be cast to '${T::class.qualifiedName}'."
-            )
-    }
+inline fun <reified T : Any> ExtensionContainer.getByName(name: String): T =
+    tryCastExtension(getByName(name), name)
+
+
+/**
+ * Looks for the extension of a given name and casts it to the expected type [T].
+ *
+ * If none found `null` is returned.
+ * If the extension is found but cannot be cast to the expected type it will throw an [IllegalStateException].
+ *
+ * @param name extension name
+ * @return extension or `null`
+ * @throws [IllegalStateException] When the given extension cannot be cast to the expected type.
+ * @since 7.2
+ */
+@Suppress("extension_shadowed_by_member")
+@Incubating
+inline fun <reified T : Any> ExtensionContainer.findByName(name: String): T? =
+    findByName(name)?.let { tryCastExtension(it, name) }
+
+
+/** Casts given [extension] to the specified type [T] or throws [IllegalStateException]. */
+@PublishedApi
+internal
+inline fun <reified T : Any> ExtensionContainer.tryCastExtension(extension: Any, name: String): T =
+    extension as? T
+        ?: throw IllegalStateException(
```

Comment:
Can I use here function `error("...")` instead of `throw IllegalStateException`? Not seen this function in the project source code.